crates/xtask-bump-check/src/xtask.rs

```diff
@@ -0,0 +1,418 @@
+    if let Some(values) = args.get_many::<String>("unstable-features") {
+        unstable_flags.extend(values.cloned());
+    }
+    let mut config_args = vec![];
+    if let Some(values) = args.get_many::<String>("config") {
+        config_args.extend(values.cloned());
+    }
+    config.configure(
+        verbose,
+        quiet,
+        color,
+        frozen,
+        locked,
+        offline,
+        &None,
+        &unstable_flags,
+        &config_args,
+    )?;
+    Ok(())
+}
+
+/// Main entry of `xtask-bump-check`.
+///
+/// Assumption: version number are incremental. We never have point release for old versions.
+fn bump_check(args: &clap::ArgMatches, config: &mut cargo::util::Config) -> CargoResult<()> {
+    let ws = args.workspace(config)?;
+    let repo = git2::Repository::open(ws.root())?;
+    let base_commit = get_base_commit(config, args, &repo)?;
+    let head_commit = get_head_commit(args, &repo)?;
+    let referenced_commit = get_referenced_commit(&repo, &base_commit)?;
+    let changed_members = changed(&ws, &repo, &base_commit, &head_commit)?;
+    let status = |msg: &str| config.shell().status(STATUS, msg);
+
+    status(&format!("base commit `{}`", base_commit.id()))?;
+    status(&format!("head commit `{}`", head_commit.id()))?;
+
+    let mut needs_bump = Vec::new();
+
+    check_crates_io(config, &changed_members, &mut needs_bump)?;
+
+    if let Some(referenced_commit) = referenced_commit.as_ref() {
+        status(&format!("compare against `{}`", referenced_commit.id()))?;
+        for referenced_member in checkout_ws(&ws, &repo, referenced_commit)?.members() {
+            let Some(changed_member) = changed_members.get(referenced_member) else {
+                let name = referenced_member.name().as_str();
+                log::trace!("skipping {name}, may be removed or not published");
+                continue;
+            };
+
+            if changed_member.version() <= referenced_member.version() {
+                needs_bump.push(*changed_member);
+            }
+        }
+    }
+
+    if !needs_bump.is_empty() {
+        needs_bump.sort();
+        needs_bump.dedup();
+        let mut msg = String::new();
+        msg.push_str("Detected changes in these crates but no version bump found:\n");
+        for pkg in needs_bump {
+            writeln!(&mut msg, "  {}@{}", pkg.name(), pkg.version())?;
+        }
+        msg.push_str("\nPlease bump at least one patch version in each corresponding Cargo.toml.");
+        anyhow::bail!(msg)
+    }
+
+    // Tracked by https://github.com/obi1kenobi/cargo-semver-checks/issues/511
+    let excludes = |cmd: &mut ProcessBuilder| {
+        cmd.arg("--exclude")
+            .arg("cargo-credential-1password")
+            .arg("--exclude")
+            .arg("cargo-credential-gnome-secret")
+            .arg("--exclude")
+            .arg("cargo-credential-macos-keychain")
+            .arg("--exclude")
+            .arg("cargo-credential-wincred");
+    };
```

Comment:
nit: couldn't his be a `let exclude_args = [...];` and then later do `cmd.args(exclude_args)`?

When I see closures being used for code reuse, I tend to look for other ways (which I generally find simpler)

Reply:
Okay I ended up removing all closures. No idea how I got obsessed by closures…